### DISCUSSION TITLE
API to get seller name

### DISCUSSION THREAD

**Discussion by NetRushJacob** [amzn/selling-partner-api-models#3222] [General] [answered]
Am I missing an API that can return the seller name? 
I see the Sellers API, but the response doesn't appear to have the name of the seller.
Any Offer Change notifications returns the seller Id, but no name.

Amazon Advertising has the seller name, so we've added it to our data models on that side. We'd like to keep things linked and accurate to what Amazon is stating the name is. I figured I'd ask here before we go down the road of a custom set of functionalities to map everything to 'the best possible name'.

**Comment by masterxml** [comment]
@NetRushJacob Hi, can you tell me how to get the seller name by Amazon Advertising? I've never used the Amazon Advertising API, but I'd like to integrate it to get the Name. Thank you.

**Reply by NetRushJacob** [reply]
https://advertising.amazon.com/API/docs/en-us/setting-up/overview
Specifically the accountInfo section of the listProfiles response https://advertising.amazon.com/API/docs/en-us/reference/2/profiles#/Profiles/listProfiles I believe (I didn't create the code that our systems use)

**Reply by masterxml** [reply]
Thanks a lot, very helpful!

**Comment by chapmanjw** [comment]
@NetRushJacob Hi Jacob, 

There is currently no SP-API for retrieving this information. Primarily for listings use-cases, we are looking at the gaps in "context" information for integrators. For example, not knowing the available vendor codes for a given vendor account. 

Beyond vendor codes and account names, are there other contextual elements that your use-cases need that are not currently provided via API today? (And particularly for managing seller or vendor listings?)

Thanks!

**Reply by NetRushJacob** [reply]
From a perspective of user experience, being able to get the seller name (likely in the Sellers API) would allow blind signups (IE seller goes to our website, signs up and starts using the services, with no contact with us) to gather all the information our systems need to give them a solid experience.

From a perspective of reporting, having the ability to get seller contact information: name, contact email (even Amazon-masked), etc could help facilitate communication between sellers.  We can get what marketplaces they can target, but we don't know who they are.

**Hypothetically**, the ability for sellers on a listing to communicate could lead to less work for Amazon. If we can communicate and collaborate on content, variation, etc, we could streamline all contributions for a given listing, making less work for Catalog to determine 'best content' as well as less support cases to 'fix' a given listing due to title/etc being changed.

**Reply by coatezy** [reply]
@NetRushJacob you have hit the nail on the head, this information would be invaluable in allowing us all to build tools with a frictionless onboarding experience. 👌 Did you manage to work around the current SP limitations?

**Reply by NetRushJacob** [reply]
@coatezy  We have no SP API workaround for this. Advertising API has it, but that would require the seller to have an advertising account AND give your application authorization to it (a second auth workflow and action as they are not combined).  We are currently working towards this as the 'solution', as our application has functionalities in both SP and Advertising.

**Reply by coatezy** [reply]
@NetRushJacob thanks, I’ve come to the same conclusion. I’m sure there are reasons but I find it baffling that there is no reference to the seller id or name.

**Comment by jamiethompson** [comment]
It is possible (at least for European sellers) to obtain a seller name by scraping the html of the page https://www.amazon.co.uk/sp?marketplaceID={marketplaceID}&seller={sellerID}. This works on amazon.co.uk, amazon.fr, amazon.de, etc.

**Edit:** I had a quick look at the equivalent page on amazon.com is https://www.amazon.com/sp?seller={sellerID}

The seller name is quite neatly obtainable from the element `h1#sellerName-rd`

It goes without saying that this solution requires a fallback in case of future changes to the html

**Reply by 0xs3n4** [reply]
Very good friend, thanks...

**Comment by NetRushJacob** [comment]
@jamiethompson 
To my understanding, that violates 4.3 of the Acceptable Use Policy: https://sellercentral.amazon.com/mws/static/policy?documentType=AUP&locale=en_US
4.3 Do not use, offer, or promote external (non-Amazon) data services that vend Information or data retrieved from Amazon's websites. 

Particularly the 'offer' in this situation as us using this information would be offering data services that vend information...

**Reply by jamiethompson** [reply]
You're absolutely correct that it does.

**Comment by LukasChristoph** [comment]
Are there any updates to the SP API in regard to obtain some sort of Seller information?
Our Application uses both the SP API & ADS API for Analytics. The Seller/Vendor authorises both Apis through the corresponding links we provide. But we have to connect both api connections internally. 

The ADS API provides (user_id / name / email) through obtain Customer Profile Information (scope profile).
The SP API provides (selling_partner_id) through Step 3. Amazon sends you the authorization information.

We dont want to rely on scraping some Customer Name from the Amazon Frontend. We just need one id/name that is available on both Apis. 
Maybe add scope profile to SP API? 
Any suggestions?

**Comment by haoxi911** [comment]
Any update on this. I don't want to scrap the store name from Amazon frontend, and we don't have an advertising account. How can we fetch the name of the store by seller ID?

**Comment by DearRude** [comment]
Any update on this. I don't want to scrap the store name from Amazon frontend, and we don't have an advertising account. How can we fetch the name of the store by seller ID?

**Comment by lombervid** [comment]
Any update on this? :/

**Comment by stuart-so** [comment]
It looks like Amazon is trying to work on this. They have added a getAccount endpoint to the sellers api. Unfortunately the endpoint is only for their sandbox api and exclusive to the EU marketplaces.

https://developer-docs.amazon.com/sp-api/docs/sellers-api-v1-reference#getaccount

**Comment by leonardonakagawa** [comment, marked as answer]
Hi all! 

As @stuart-so mentioned, we released the getAccount [operation](https://developer-docs.amazon.com/sp-api/docs/sellers-api-v1-reference#operations) which is currently in beta phase in sandbox for the EU marketplaces which return the seller business information including the seller name. Please subscribe to our RSS feeds to receive last updates regarding SP-API when this gets general available in production: 

https://developer-docs.amazon.com/sp-api/changelog.rss

To Integrate RSS Feeds on Slack you can check our video: [Integrate SP-API RSS feeds into Slack
](https://www.youtube.com/watch?v=WT9miRiMA2g&t=17s)


Thanks,
Leonardo Nakagawa
Solutions Architect, Selling Partner API Developer Support

**Reply by rabbiveesh** [reply]
Is there a reason that it's only in EU? Does that mean it's not gonna be rolled out elsewhere?

**Comment by ericblade** [comment]
Unbelieveable that you've added a getAccount and it doesn't include the "sellerId" information, which is a requirement to other API calls, and is unobtainable except through a completely unrelated API for getting pricing fees.
